### PR TITLE
CMP-2582: Update CO install documentation to include ROSA support

### PR DESCRIPTION
**Pull request by rhmdnd**
The next Compliance Operator release (1.5.0) will include ROSA HCP support.

This commit updates the installation documentation show users how they can install the operator if they're running ROSA HCP clusters.

Version(s):

Link to docs preview:

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/compliance-operator-rosa-installation.adoc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+// * security/compliance_operator/co-management/compliance-operator-installation.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installing-compliance-operator-cli_{context}"]
+= Installing the Compliance Operator on ROSA
+
+As of the Compliance Operator 1.5.0 release, the operator is tested against Red Hat
+OpenShift Service on AWS (ROSA) using hosted control plane topologies (HCP).
+
+ROSA HCP clusters restrict access to the control plane, which is managed by Red
```
Comment (xiaojiey) [inline modules/compliance-operator-rosa-installation.adoc:12]: Maybe you want to say
`ROSA HCP clusters **have** restrict access to the control plane, which is managed by Red Hat`

File: security/compliance_operator/co-management/compliance-operator-installation.adoc [hunk 2]
```diff
@@ -10,7 +10,7 @@ Before you can use the Compliance Operator, you must ensure it is deployed in th
 
 [IMPORTANT]
 ====
-The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated, Red Hat OpenShift Service on AWS, and Microsoft Azure Red Hat OpenShift. For more information, see the link:https://access.redhat.com/solutions/6983418[Red Hat Knowledgebase Solution #6983418].
+The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated and Microsoft Azure Red Hat OpenShift. For more information, see the link:https://access.redhat.com/solutions/6983418[Red Hat Knowledgebase Solution #6983418].
```
Comment (xiaojiey) [inline security/compliance_operator/co-management/compliance-operator-installation.adoc:13]: @rhmdnd I am wondering if we still have the same issue on classic ROSA - Red Hat OpenShift Service on AWS. Should be keep the "Red Hat OpenShift Service on AWS"?

Comment (rhmdnd) [reply security/compliance_operator/co-management/compliance-operator-installation.adoc]: Good call - fixed.

File: modules/compliance-operator-rosa-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * security/compliance_operator/co-management/compliance-operator-installation.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="installing-compliance-operator-rosa_{context}"]
+= Installing the Compliance Operator on ROSA hosted control planes (HCP)
+
+As of the Compliance Operator 1.5.0 release, the operator is tested against Red Hat
+OpenShift Service on AWS (ROSA) using hosted control plane topologies (HCP).
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:10]: * Capitalize "Operator" per [repo term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#operator).
* There's a [common-attribute](https://github.com/openshift/openshift-docs/blob/main/_attributes/common-attributes.adoc?plain=1#L164) available for the full ROSA name.
```suggestion
As of the Compliance Operator 1.5.0 release, the Operator is tested against {product-rosa} using hosted control plane topologies (HCP).
```
There are also some for HCP, if you like:
```
// Hosted control planes related attributes
:hcp-capital: Hosted control planes
:hcp: hosted control planes
```

File: modules/compliance-operator-rosa-installation.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+ROSA HCP clusters have restricted access to the control plane, which is managed
+by Red Hat. By default, the Compliance Operator will schedule to nodes within
+the `master` node pool, which isn't available in ROSA HCP installations. This
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:14]: Avoid contractions per [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#contractions):
```suggestion
the `master` node pool, which is not available in ROSA HCP installations. This
```

File: modules/compliance-operator-rosa-installation.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+requires you to configure the `Subscription` object in a way that allows the
+operator to schedule on available node pools. This step is necessary for a
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:16]:
```suggestion
Operator to schedule on available node pools. This step is necessary for a
```

File: modules/compliance-operator-rosa-installation.adoc [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+.Prerequisites
+
+* You must have `admin` privileges.
+
+.Procedure
+
+. Define a `Namespace` object:
++
+.Example `namespace-object.yaml`
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:27]:
```suggestion
.Example `namespace-object.yaml` file
```
I won't comment on all of these, but apply consistently like this for any similar block titles.

File: modules/compliance-operator-rosa-installation.adoc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    openshift.io/cluster-monitoring: "true"
+    pod-security.kubernetes.io/enforce: privileged <1>
+  name: openshift-compliance
+----
+<1> in {product-title} {product-version}, the pod security label must be set to `privileged` at the namespace level.
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:38]:
```suggestion
<1> In {product-title} {product-version}, the pod security label must be set to `privileged` at the namespace level.
```

File: modules/compliance-operator-rosa-installation.adoc [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+  channel: "stable"
+  installPlanApproval: Automatic
+  name: compliance-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/worker: "" <1>
+----
+<1> update the operator deployment to deploy on `worker` nodes.
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:89]:
```suggestion
<1> Update the Operator deployment to deploy on `worker` nodes.
```

File: modules/compliance-operator-rosa-installation.adoc [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+. Create the `Subscription` object:
++
+[source,terminal]
+----
+$ oc create -f subscription-object.yaml
+----
+
+.Verification
+
+. Verify the installation succeeded by inspecting the CSV file:
```
Comment (adellape) [inline modules/compliance-operator-rosa-installation.adoc:100]:
```suggestion
. Verify the installation succeeded by inspecting the cluster service version (CSV) object:
```

File: security/compliance_operator/co-management/compliance-operator-installation.adoc [hunk 10]
```diff
@@ -10,7 +10,7 @@ Before you can use the Compliance Operator, you must ensure it is deployed in th
 
 [IMPORTANT]
 ====
-The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated, Red Hat OpenShift Service on AWS, and Microsoft Azure Red Hat OpenShift. For more information, see the link:https://access.redhat.com/solutions/6983418[Red Hat Knowledgebase Solution #6983418].
+The Compliance Operator might report incorrect results on managed platforms, such as OpenShift Dedicated, Red Hat OpenShift Service on AWS Classic, and Microsoft Azure Red Hat OpenShift. For more information, see the link:https://access.redhat.com/solutions/6983418[Red Hat Knowledgebase Solution #6983418].
```
Comment (adellape) [inline security/compliance_operator/co-management/compliance-operator-installation.adoc:13]: Realize this is older content, but consider the format of the KBase link per https://redhat-documentation.github.io/supplementary-style-guide/#rh-kb-links.

**Comment by adellape** [comment]
Also reminder to indicate in PR description which branch versions this should be cherry-picked to.

**Comment by GroceryBoyJr** [comment]
Please note: https://github.com/openshift/openshift-docs/pull/77523 is a duplcate of this PR with which I am making the corrections and pushing the update into the documentation. This PR will be closed after [PR 77523](https://github.com/openshift/openshift-docs/pull/77523) is merged.

**Pull request closed without merging**